File: app/src/main/java/com/boardgamegeek/model/PlayDeleteResponse.java
package com.boardgamegeek.model;

import androidx.annotation.NonNull;

import okhttp3.OkHttpClient;
import okhttp3.Request;

public class PlayDeleteResponse extends PlayPostResponse {
	private PlayDelete playDelete;

	public PlayDeleteResponse(@NonNull OkHttpClient client, @NonNull Request request) {
		super(client, request);
	}

	@Override
	protected void saveContent(String content) {
		playDelete = gson.fromJson(content, PlayDelete.class);
	}

	public boolean isSuccessful() {
		return !(hasError() || playDelete == null) && playDelete.isSuccessful();
	}

	private static class PlayDelete {
		@SuppressWarnings("unused") private boolean success;

		public boolean isSuccessful() {
			return success;
		}
	}
}

File: app/src/main/java/com/boardgamegeek/service/PlayDeleteResponse.kt
package com.boardgamegeek.service

import okhttp3.OkHttpClient
import okhttp3.Request

class PlayDeleteResponse(client: OkHttpClient, request: Request) : PlayPostResponse(client, request) {
    private var playDelete: PlayDelete? = null

    override fun saveContent(content: String?) {
        playDelete = gson.fromJson(content, PlayDelete::class.java)
    }

    val isSuccessful: Boolean
        get() = !(hasError()) && playDelete?.success == true

    private class PlayDelete {
        val success = false
    }
}